src/group.rs

```diff
@@ -6,19 +6,43 @@ use crate::{
     internal::{group_api, user_api::UserId},
     Result,
 };
+use std::{collections::HashSet, iter::FromIterator};
 use tokio::runtime::current_thread::Runtime;
 
+// This is used for GroupCreateOpts that have been standardized with the GroupCreateOpts::standardize function.
+// `add_as_member` and `add_as_admin` have been removed, with the calling user added to the `members` and `admins` lists.
+// `all_users` holds all the users who need their public keys looked up with duplicates removed.
+struct GroupCreateOptsStd {
+    id: Option<GroupId>,
+    name: Option<GroupName>,
+    owner: Option<UserId>,
+    admins: Vec<UserId>,
+    members: Vec<UserId>,
+    all_users: Vec<UserId>,
+    needs_rotation: bool,
+}
+
 #[derive(Clone)]
 /// Options for group creation.
 pub struct GroupCreateOpts {
     // unique id of a group within a segment. If none, the server will assign an id.
     id: Option<GroupId>,
     // human readable name of the group. Does not need to be unique.
     name: Option<GroupName>,
+    // true (default) - creating user will be added as an admin of the group.
+    // false - creating user will not be added as an admin of the group.
+    add_as_admin: bool,
     // true (default) - creating user will be added to the group's membership (in addition to being the group's admin);
     // false - creating user will not be added to the group's membership
     add_as_member: bool,
-    // list of users to add as members to the group.
+    // None (default) - The creating user will be the owner of the group.
+    // Some(UserId) - The provided user will be the owner of the group.
+    // Note that the owner must be included in the `admins` list.
```

Comment:
Should we maybe just take the user ID that was provided here and manually add it to the list of admins so that they don't have to specify it in two places?

Reply:
So that's actually what we are doing, I just have it documented weirdly. We decided on a semi-lenient way of requiring that the owner is in the admin list. If:
- `owner` is Some(UserId), I put the UserId in the admins list for them because they were explicit about choosing an owner.
- `owner` is None (which makes the calling user the owner), then they're hopefully using `add_as_admin` as well. If not, they did something wrong and are being confusing about what they want, so it will error on them for not being in the admins list. 

So we're kind of in a half-strict, half-lenient state where if they're explicit about the owner, we do it for them, but if they just leave it at None, they have to do it themselves (or with `add_as_admin`).
I'm open to suggestions about changing the documentation or implementation, this is just what we've decided on for now.

Reply:
Ok, I think the logic is fine, I'd just change this wording to something like

```
    // Specifies who the owner of this group is. Group owners have the same permissions as other admins but they cannot be removed as admins from this group.
    // None (default) - The creating user will be the owner of the group. Cannot be used if `add_as_admin` is set to false as the group must have an owner.
    // Some(UserId) - The provided user will be the owner of the group. This ID will automatically be added to the admins list.
```

Reply:
Thanks for the feedback. I've added in this distinction to make it more clear.